libraries/Ethernet/src/ETH.cpp

```diff
@@ -18,6 +18,8 @@
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#ifdef CONFIG_ETH_ENABLED
```

Comment:
you are checking too early if `CONFIG_ETH_ENABLED` is defined. At this point in the header, nothing is yet included. You need to at least `#include "sdkconfig.h"` for this to work